e2e_tests/tests/test_system.py

```diff
@@ -485,6 +485,16 @@ def test_fail_on_first_validation() -> None:
     )
 
 
+@pytest.mark.e2e_cpu  # type: ignore
```

Comment:
Can we make this into a unit test rather than an e2e test?  We already have some infrastructure in `trial_workload_sequencer_test.go`.

Reply:
What about both? In theory, this covers more than just the trial workload sequencer. This covers making sure this class of failures is handled by the entire system.

Reply:
Works for me.

Reply:
cool, done